RecoEcal/EgammaCoreTools/interface/DeepSCGraphEvaluation.h

```diff
@@ -0,0 +1,64 @@
+#ifndef RecoEcal_EgammaCoreTools_DeepSCGraphEvaluation_h
+#define RecoEcal_EgammaCoreTools_DeepSCGraphEvaluation_h
+
+#include "PhysicsTools/TensorFlow/interface/TensorFlow.h"
+#include <vector>
+#include <array>
+#include <memory>
+#include <string>
+#include <functional>
+
+//author: Davide Valsecchi
+//description:
+// Handles Tensorflow DNN graphs and variables scaler configuration.
+// To be used for DeepSC.
+
+namespace reco {
+
+  struct DeepSCConfiguration {
+    std::string modelFile;
+    std::string scalerFileClusterFeatures;
+    std::string scalerFileWindowFeatures;
+    uint nClusterFeatures;
+    uint nWindowFeatures;
+    static constexpr uint nRechitsFeatures = 4;
+    uint maxNClusters;
+    uint maxNRechits;
+    uint batchSize;
+    std::string collectionStrategy;
+  };
+
+  struct DeepSCInputs {
+    std::vector<std::vector<std::vector<double>>> clustersX;
```

Comment:
I have included more documentation for it. Do you suggest to simplify it by producing a vector of "inputs" where each entry is a single "window", instead of already having a full nested structure?

Reply:
I didn't really think what is a better approach here (specific classes maybe?). But the additional documentation is welcome and I'm fine to leave it like that.

Reply:
Sorry, but this **is** horrible coding practice.

Data should be organised in data structures, not vectors of vectors of vectors, whose "interface" (i.e. the meaning of the parameters in each position and level of the vectors) arbitrarily change, without any compile-time checks,

Reply:
Also a memory overhead nightmare (especially as push_back is used with no reserve in places). There are also maps of vectors and functions that return these instead heavyweight objects instead of handling them via function arguments..

Reply:
@valsdav, I doesn't look like these comments have been addressed at all ?